spec/models/service_template_spec.rb

```diff
@@ -50,20 +50,24 @@
       false_expression = MiqExpression.new("=" => {"field" => "ServiceTemplate-name", "value" => "bar"})
       FactoryGirl.create(:custom_button,
                          :name                  => "visible button",
-                         :applies_to_class      => "Service",
+                         :applies_to_class      => "ServiceTemplate",
+                         :applies_to_id         => service_template.id,
                          :visibility_expression => true_expression)
       FactoryGirl.create(:custom_button,
                          :name                  => "hidden button",
-                         :applies_to_class      => "Service",
+                         :applies_to_class      => "ServiceTemplate",
+                         :applies_to_id         => service_template.id,
                          :visibility_expression => false_expression)
-      FactoryGirl.create(:custom_button_set).tap do |group|
+      service_template.custom_button_sets << FactoryGirl.create(:custom_button_set).tap do |group|
```

Comment:
Similarly - this custom button set is generic and not to be associated with a particular record.

Reply:
@imtayadeway would you recommend to add  another test to demonstrate use of generic buttons to control visibility ?

Reply:
@imtayadeway  changed to use generic buttons